tests/Unit/Projects/Applications/Templates/HtmlTest.php

```diff
@@ -0,0 +1,150 @@
+use Servidor\Projects\Project;
+use Tests\TestCase;
+
+class HtmlTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    public function document_root_defaults_to_system_dir(): void
+    {
+        $app = new Application([
+            'template' => 'html',
+            'source_provider' => 'github',
+            'source_repository' => 'dshoreman/servidor-test-site',
+        ]);
+        $project = Project::create(['name' => 'htmlroot']);
+        $project->applications()->save($app);
+
+        $this->assertEquals('/var/www/htmlroot/servidor-test-site', $app->document_root);
+    }
+
+    /** @test */
+    public function enabling_project_creates_config_symlink(): Application
+    {
+        $this->assertFileNotExists($link = '/etc/nginx/sites-enabled/symlinkery.dev.conf');
+
+        $project = Project::create(['name' => 'symlinkery', 'is_enabled' => true]);
+        $project->applications()->save($app = new Application([
+            'domain_name' => 'symlinkery.dev',
+            'source_repository' => 'dshoreman/servidor-test-site',
+            'source_provider' => 'github',
+            'source_branch' => 'develop',
+        ]));
+
+        $this->assertFileExists($link);
+        $this->assertTrue(is_link($link));
+        $this->assertEquals('/etc/nginx/sites-available/symlinkery.dev.conf', readlink($link));
+
+        return $app;
+    }
+
+    /**
+     * @test
+     * @depends enabling_project_creates_config_symlink
+     */
+    public function enabling_project_does_not_create_symlink_when_already_valid(Application $app): void
+    {
+        $linkBefore = readlink($link = '/etc/nginx/sites-enabled/symlinkery.dev.conf');
+
+        $app->source_branch = 'master';
+        $app->save();
+
+        $this->assertEquals('master', $app->source_branch);
+        $this->assertSame($linkBefore, readlink($link));
+    }
+
+    /**
+     * @test
+     * @depends enabling_project_creates_config_symlink
+     */
+    public function outdated_symlinks_get_replaced(Application $app): void
+    {
+        $link = '/etc/nginx/sites-enabled/symlinkery.dev.conf';
+        exec("sudo rm ${link} && sudo ln -s /dev/null ${link}");
+        $this->assertNotEquals($link, readlink($link));
+
+        $app->source_branch = 'develop';
+        $app->save();
+
+        $this->assertFileExists($link);
+        $this->assertFileExists($vhost = '/etc/nginx/sites-available/symlinkery.dev.conf');
+        $this->assertTrue(is_link($link));
+        $this->assertEquals($vhost, readlink($link));
+    }
+
+    /**
+     * @test
+     * @depends enabling_project_creates_config_symlink
+     */
+    public function disabling_project_removes_nginx_symlink(Application $app): void
+    {
+        $this->assertFileExists('/etc/nginx/sites-enabled/symlinkery.dev.conf');
+
+        $app->project->is_enabled = false;
+        $app->save();
+
+        $this->assertFileNotExists('/etc/nginx/sites-enabled/symlinkery.dev.conf');
```

Comment:
The file *is* removed via the observer, but the `file_exists` call made by phpunit is still returning `true`. This is probably an issue of caching by PHP.

Clearing the status cache between removal and testing *should* solve that.

```suggestion
        $app->project->is_enabled = false;
        $app->save();

        clearstatcache();
        $this->assertFileNotExists('/etc/nginx/sites-enabled/symlinkery.dev.conf');
```